### DISCUSSION TITLE
Discussion board improvements

### DISCUSSION THREAD

**Discussion by dave-csc** [apache/hop#4676] [Ideas]
Hello,

I'd like to propose some (more or less feasible) improvements to this discussion board:

- **Show closed discussions by default:** you currently need to remove a filter to show them. Sometimes a discussion results in being hidden even if it's actually resolved ([here's an example](https://github.com/apache/hop/discussions/4661)) and could be useful to other users, too
- **Enforce a "mark as answered" policy, and allow only moderators to close discussions:** for the same reason as above. In this way closed discussions can indeed be hidden, since a moderator could have judged them as "trash"
- **Review the link to the mailing list:** currently every single action on a post produces an email sent to the Hop users mailing list, even a simple correction of a typo or a message deletion

**Comment by hansva** [comment]
Hi @dave-csc ,

Though I agree with the first point, it seems GitHub discussions do not have an option to avoid closing discussions.
For Q&A it does seem best to keep discussions open and mark the Answer. This brings us to the second point the original creator and collaborators can mark the discussion as answered. I think we should encourage the original creators to mark the answer that provided the best feedback as the "Answer". Having collaborators mark answers is possible but should be avoided.

We can pin a discussion at the top of each or all categories explaining what is expected of the users.

For the final point, we can indeed limit the mails a bit. the events that trigger a mail are listed in the [.asf.yaml](https://github.com/apache/hop/blob/main/.asf.yaml) file located at the root of this repository. I suggest we remove the following events:

- edit_discussion
- edit_comment_discussion
- delete_comment_discussion

**Comment by dave-csc** [comment]
Hi @hansva,

thanks for the feedback, as mentioned I had no idea about the feasibility of my suggestions.

I raise another issue regarding the duplication of discussions in the mailing list (ML): some users are answering to the posts in mailing list, and those replies aren't included in the discussion board (examples here [github](https://github.com/apache/hop/discussions/4678#discussioncomment-11505218)/[ML](https://lists.apache.org/thread/ctjnk2df5of8xxt987qb2zyfq5bnkkfo) and here [github](https://github.com/apache/hop/discussions/4683)/[ML](https://lists.apache.org/thread/x2n6xtsy52qo3ljxgxs3mt75pqb4t6jc))

**Reply by hansva** [reply]
Unfortunately, this mail traffic is one-way only 2 possible solutions:
- move the discussion to another mailing list
- educate our users

I like that these discussions end up in the user mailing list, so I am not a fan of option 1

**Reply by hansva** [reply]
I created #4695 to remove some of the mails

**Reply by hansva** [reply]
I am going to revert this change, it seems the mails are sent anyway. it's just the subject format that changes causing the threads to combine.
